pkg/kubectl/cmd/portforward.go

```diff
@@ -76,11 +75,21 @@ func (*defaultPortForwarder) ForwardPorts(req *client.Request, config *client.Co
 
 func RunPortForward(f *cmdutil.Factory, cmd *cobra.Command, args []string, fw portForwarder) error {
 	podName := cmdutil.GetFlagString(cmd, "pod")
-	if len(podName) == 0 {
-		return cmdutil.UsageError(cmd, "POD_NAME is required for port-forward")
+	if len(podName) == 0 && len(args) == 0 {
```

Comment:
No,  it doestn't. We should check len(podName) and len(args) firstly, the other way to check is: 
        if len(podName) != 0 {
		printDeprecationWarning("port-forward POD", "-p POD")
	} else if  len(args) == 0 {
                return cmdutil.UsageError(cmd, "POD is required for port-forward")
        } else {
		podName = args[0]
		args = args[1:]
	}
        But I don't think this is better than the original one, so I keep the original one